docs/modules/ROOT/pages/explanations/decisions/api-design.adoc

```diff
@@ -0,0 +1,250 @@
+The set of available features is most likely not the same between most providers.
+Also with Crossplane we're a bit limited on what's possible with translating the source API spec into the target API spec:
+
+* No conditionals in compositions
+* Only a limited set of transformations
+* Can't create random values
+
+[NOTE]
+This is not a complete list of limitations.
+
+== Requirements
+
+* APIs for a set of similar services feel similar
+* Only minimal configuration is needed for a default case
+* Unique features of a service provider can be leveraged
+
+== Proposals
+
+=== Proposal 1: Use same XRD with provider specific fields
+
+There's exactly one XRD for a set of services.
+This XRD contains a minimal set of common parameters, as well as a field for provider specific configurations.
+
+.Example Claim
+[source, yaml]
+----
+apiVersion: appcat.vshn.io/v1
+kind: PostgreSQL
+metadata:
+  name: my-instance
+spec:
+  parameters:
+    name: my-instance
+    majorVersion: '14'
+    pgSettings: {}
+    exoscale:
+      plan: hobbyist-2
+      type: pg
+      maintenance:
+        day: 2
+        hour: 22
+        minute: 0
+      backup:
+        hour: 20
+        minute: 0
+    vshn:
+      minorVersion: '4'
+----
+
+Advantages::
+
+* We can use the same XRD
+* Adding new providers to the list is easy
+
+Disadvantages::
+
+* We don't really have a "common look"
+* The customer needs to know what the underlying service provider is, even for simple instances
+
+=== Proposal 2: Separate XRDs
+
+This approach will have different XRDs for each provider.
+However, they will follow the same structure, as far as possible.
+All the toplevel fields will be the same, the differences will be apparent in the sub-fields.
+They will also have the same kind, but a different APIVersion.
+
+.Example Claim
+[source, yaml]
+----
+# Exoscale example
+apiVersion: exoscale.appcat.vshn.io/v1
+kind: PostgreSQL
+metadata:
+  name: my-instance
+spec:
+  parameters:
+    name: my-instance
+    service:
+      version: '14'
+      type: pg
+      settings: {}
+    size:
+      plan: hobbyist-2
+    maintenance:
+      day: 2
+      hour: 22
+      minute: 0
+    backup:
+      hour: 20
+      minute: 0
+
+# Converged example
+apiVersion: vshn.appcat.vshn.io/v1
+kind: PostgreSQL
+metadata:
+  name: my-instance
+spec:
+  parameters:
+    name: my-instance
+    service:
+      version: '14.4'
+      settings: {}
+      pgbouncer:
+        settings: {}
+    size:
+      cpu: 2
+      memory: 8Gi
+      disk: 200Gi
+      encrypted: true
+    maintenance:
+      day: 2
+      hour: 22
+      minute: 0
+    backup:
+      hour: 20
+      minute: 0
+----
+
+Advantages::
+
+* Same look between all providers, deploying PostgreSQL to any backend feels familiar
+* Specifics can be handled in a sensible way
+
+Disadvantages::
+
+* The user could get confused by objects with the same kind
+
+=== Proposal 3: Composed XRDs
+
```

Comment:
tbh I would remove this proposal as its just not a real option.
Having a base CRDs just to generalize 3 parameters is, IMO, unacceptable for both the user and maintainers.

Reply:
I'd like to keep it in the page, just for completeness' sake.